CODINGSTYLE.md

```diff
@@ -0,0 +1,335 @@
+Similarly, you should not use braces for the body of a `while`/`if`/`for` that is a one-liner (but you should if it is a two-liner `if/else`).
+
+```scala
+def abs(x: Int): Int =
+  if (x >= 0) x
+  else -x
+
+def clamp(v: Int, min: Int, max: Int): Int = {
+  if (v < min) min
+  else if (v > max) max
+  else v
+}
+```
+
+Every expression that spans more than one line should be enclosed in braces.
+For example, this is wrong:
+
+```scala
+val iterator =
+  new Iterator[A] {
+    ...
+  }
+```
+
+It should be:
+
+```scala
+val iterator = {
+  new Iterator[A] {
+    ...
+  }
+}
+```
+
+#### Braces in lambdas
+
+In lambdas (anonymous functions), the opening brace must be placed before the formal arguments, and not after the `=>`:
+
+```scala
+val f = { (x: Int) =>
+  body
+}
+```
+
+### Spaces
+
+There must not be any space before the following tokens: `: , ; )`
+
+There must be exactly one space after the following tokens: `: , ; if for while`
+Sometimes, it is acceptable to have several spaces, for vertical alignment reasons.
+
+There must be exactly one space before and after `{` and `}`.
+With the exception of partial import, where there is no space on either side.
+
+Binary operators, including `=>`, must have a single space on both sides.
+Sometimes, spaces can be removed to highlight the relatively higher priority wrt. to a neighboring operator, for easier visual parsing.
+For example, instead of `x < len - 1`, it is better to write `x < len-1`, highlighting that `-` has a higher priority than `<`.
+
+Unary operators must not be followed by a space.
+
+### Method call style
+
+Usually, parentheses should be used for actual parameters to a method call.
+Braces should be used instead if an argument list has only a lambda, and that lambda does not fit in an inline one-liner.
+
+In general, dot-notation should be used for non-symbolic methods, and infix notation should be used for symbolic methods.
+Infix notation is also used if the only argument is a brace lambda.
+
+Examples:
+
+```scala
+// inline lambda, hence (), hence dot-notation
+list.map(x => x * 2)
+
+// long lambda, hence braces, hence infix notation
+list map { x =>
+  if (x < 5) x
+  else x * 2
+}
+
+// symbolic operator, hence infix notation
+value :: list
+```
+
+When calling a method declared with an empty pair of parentheses, use `()`, except if the method is Java-defined *and* does not have side-effects.
+
+### Method definition
+
+All public and protected methods must have an explicit result type.
+
+Procedure syntax must not be used. `: Unit =` must be used instead.
+
+Side-effect-free methods without formal parameters should be declared without `()`, unless either a) it overrides a methode defined with `()` (such as `toString()`) or b) it implements a Java method in the Java libraries.
+
+### for comprehensions
+
+For-comprehensions must use `()` if they have a single generator without `if`, such as:
```

Comment:
Even if the rhs is a pattern? I think our codebase violates this rule (especially me)...